ghcide.cabal

```diff
@@ -144,7 +144,7 @@ library
         Development.IDE.Plugin.CodeAction.RuleTypes
         Development.IDE.Plugin.Completions.Logic
         Development.IDE.Plugin.Completions.Types
-    if impl(ghc > 8.7) || flag(ghc-lib)
+    if (impl(ghc > 8.7) && impl(ghc < 8.10)) || flag(ghc-lib)
```

Comment:
If this is correct, then why do we need the file in tree for 8.8 as well?

Reply:
This file contains the implementation of the `mkHieFile` function. This function is present in GHC 8.10. And the implementation for GHC-8.8 is based on HIE files from GHC, where the implementation of this function for GHC 8.6 is based on `ghcide` own reimplementation of HIE files. So there are 3 cases for this function.

Reply:
The version of `mkHieFile` in GHC 8.8 was unsuitable for ghcide, so we inlined the implementation locally to fix it. After that, @mpickering yourself propagated the fix upstream, which is the reason it happily does not need to be inlined when building against GHC 8.10 :)